spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ConfigStore.java

```diff
@@ -131,21 +131,32 @@ private boolean watchedKeyValid(String watchedKey) {
      */
     public String[] getLabels() {
         if (!StringUtils.hasText(this.getLabel())) {
-            return EMPTY_LABEL_ONLY;
+            return EMPTY_LABEL;
         }
 
+        // The use of trim makes label= dev,prod and label= dev, prod equal.
         List<String> labels =  Arrays.stream(this.getLabel().split(LABEL_SEPARATOR))
-                .filter(StringUtils::hasText)
-                .map(String::trim)
+                .map(label -> mapLabel(label))
                 .distinct()
                 .collect(Collectors.toList());
-
+        
+        if (this.getLabel().endsWith(",")) {
```

Comment:
Did we describe this scenario in our doc? Looks like label ending with a comma is a special case.

Reply:
This is because I found ",MyLabel" to work, but "MyLabel," to not work consistently between yaml and properties. The empty label or '' no text as (No Label) vs Null label.